### PR TITLE
Remove Ice.Plugin.IceSSL usage

### LINKED ISSUE #2384: Remove "Ice.Plugin.IceSSL" checks in tests
We have some tests that still check for `"Ice.Plugin.IceSSL"` property, we should cleanup this to check for `"Ice.Default.Protocol" == "ssl"`

For example: 
https://github.com/zeroc-ice/ice/blob/ff285270ca57c147172e078037bede8464cb70b1/cpp/test/Ice/binding/AllTests.cpp#L799-L800

### PR DESCRIPTION
**Pull request by pepone**
Fix #2384

### CODE REVIEW

**Review by InsertCreativityHere — APPROVED** [review]
Looks good to me!

File: cpp/src/Ice/SSL/SecureTransportTransceiverI.cpp [hunk 1]
```diff
@@ -195,9 +195,9 @@ Ice::SSL::SecureTransport::TransceiverI::initialize(IceInternal::Buffer& readBuf
                 }
                 else if (_trust)
                 {
-                    for (CFIndex i = 0, count = SecTrustGetCertificateCount(_trust.get()); i < count; ++i)
+                    if (SecTrustGetCertificateCount(_trust.get()) > 0)
```
Comment (bernardnormier) [inline cpp/src/Ice/SSL/SecureTransportTransceiverI.cpp:198]: Unrelated fix?

**Review by bernardnormier — APPROVED** [review]
Looks good.

What will happen if the user keep this property in his config file? Do we have any validation of plug-in names with an Ice prefix?

File: cpp/src/Ice/SSL/SecureTransportTransceiverI.cpp (hunk 1 above)
Comment (pepone) [reply cpp/src/Ice/SSL/SecureTransportTransceiverI.cpp:198]: Yes, this was bogus we want the leaf certificate with index 0. The previous code overrides `_peerCertificate` for each cert in the chain which doesn't make sense.
